src/gui/search/searchwidget.cpp

```diff
@@ -161,6 +162,12 @@ bool SearchWidget::eventFilter(QObject *object, QEvent *event)
         {
             closeTab(tabIndex);
             return true;
+        } else if (mouseEvent->button() == Qt::RightButton)
+        {
+            QMenu m;
+            m.addAction(tr("Close tab"), [=](){ closeTab(tabIndex); return true; });
```

Comment:
I believe you can just capture `this`.

Reply:
But I need tabIndex. So i will use [this, tabIndex]